server/build.gradle

```diff
@@ -7,8 +7,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.4.2'
-        
+        classpath 'com.android.tools.build:gradle:7.0.4'
```

Comment:
@juliagarrigos are there any reasons why you chose AGP 7.0.4 and Gradle 7.2 instead of picking latest versions?

Reply:
I wanted to keep the changes in this PR limited, updating to gradle 8 was requiring extra changes that required extra investigation. I'll try to create a PR soon updating to the latest gradle version.

Reply:
In the end the changes required were not that important so I managed to include the update in this PR 💃🏻